File: benchmarks/struct_records/struct_records.swift
struct Person {
    var name: String
    var age: Int
}

func makePerson(_ i: Int) -> Person {
    Person(name: "user" + String(i), age: 20 + i % 50)
}

func birthday(_ p: Person) -> Person {
    Person(name: "user" + String(p.age), age: p.age + 1)
}

func score(_ p: Person) -> Int {
    p.name.utf8.count + p.age
}

var total = 0
for i in 0..<500000 {
    let p = makePerson(i)
    let q = birthday(p)
    total += score(q)
}
precondition(total == 25750000, "struct_records checksum")
print("assert passed, struct_records is correct")

File: benchmarks/struct_records/struct_records.rs
struct Person {
    name: String,
    age: i64,
}

fn make_person(i: i64) -> Person {
    Person {
        name: format!("user{}", i),
        age: 20 + i % 50,
    }
}

fn birthday(p: Person) -> Person {
    Person {
        name: p.name,
        age: p.age + 1,
    }
}

fn score(p: &Person) -> i64 {
    p.name.len() as i64 + p.age
}

fn main() {
    let mut total = 0i64;
    for i in 0..500000 {
        let p = make_person(i);
        let q = birthday(p);
        total += score(&q);
    }
    assert!(total == 27638890, "struct_records checksum");
    println!("assert passed, struct_records is correct");
}
